transformer_engine/pytorch/cpu_offload.py

```diff
@@ -11,6 +11,15 @@
 
 __all__ = ['get_cpu_offload_context']
 
+TORCH_MAJOR = int(torch.__version__.split(".")[0])
```

Comment:
That said I'm not quite sure what is this exactly supposed to do? One could be using torch.compile in the earlier versions of pyTorch too instead of nvFuser, so this check seems wrong.

Reply:
One reason could be, these tensors started popping in only after we started using NVFuser through torch.compile based on the version. 

Potentially we could check unconditionally, we just didn't have the need to in any of our NeMo/MCORE systems.
 
Also, even if stray tensors were there in the older version, they never went unread after saving. If we remove this check, we might miss the assert check where we ensure all tensors are read back after saving. Just didn't want some unknown errors in older version to get masked without my notice. Again, I'm leaving upto you, we can remove it if needed.

Reply:
I'd say we should remove it, but actually check for the expected types in the tensor_push function (so instead of checking if type is not Tensor, check if type is FakeTensor or FunctionalTensor.

Reply:
@ptrendx I fixed as per your suggestion. Can we merge this if rest all are good?